src/crafting.cpp

```diff
@@ -1882,3 +1913,16 @@ const std::string &recipe::ident() const
 {
     return ident_;
 }
+
+std::vector<npc *> player::get_crafting_helpers() const
+{
+    std::vector<npc *> ret;
+    for( auto &elem : g->active_npc ) {
+        if( rl_dist( elem->pos(), pos() ) < PICKUP_RANGE && elem->is_friend() &&
```

Comment:
Why do you think that? It's perfectly reasonable you should both be in the same room when working on a project. Otherwise we have situations like NPC's helping you craft despite being locked in an adjacent room.